datastore/sqlite_datastore.nim

```diff
@@ -33,53 +39,149 @@ type
     containsStmt: ContainsStmt
     deleteStmt: DeleteStmt
     env: SQLite
+    getDataCol: BoundDataCol
     getStmt: GetStmt
     putStmt: PutStmt
     readOnly: bool
 
 const
-  IdType = "TEXT"
-  DataType = "BLOB"
-  TimestampType = "INTEGER"
-
   dbExt* = ".sqlite3"
-  tableTitle* = "Store"
+  tableName* = "Store"
+
+  idColName = "id"
+  dataColName = "data"
+  timestampColName = "timestamp"
+
+  idColIndex = 0
+  dataColIndex = 1
+  timestampColIndex = 2
+
+  idColType = "TEXT"
+  dataColType = "BLOB"
+  timestampColType = "INTEGER"
 
   # https://stackoverflow.com/a/9756276
   # EXISTS returns a boolean value represented by an integer:
   # https://sqlite.org/datatype3.html#boolean_datatype
   # https://sqlite.org/lang_expr.html#the_exists_operator
   containsStmtStr = """
     SELECT EXISTS(
-      SELECT 1 FROM """ & tableTitle & """
-      WHERE id = ?
+      SELECT 1 FROM """ & tableName & """
+      WHERE """ & idColName & """ = ?
     );
   """
 
   createStmtStr = """
-    CREATE TABLE IF NOT EXISTS """ & tableTitle & """ (
-      id """ & IdType & """ NOT NULL PRIMARY KEY,
-      data """ & DataType & """,
-      timestamp """ & TimestampType & """ NOT NULL
+    CREATE TABLE IF NOT EXISTS """ & tableName & """ (
+      """ & idColName & """ """ & idColType & """ NOT NULL PRIMARY KEY,
+      """ & dataColName & """ """ & dataColType & """,
```

Comment:
We already have tests for storage and retrieval of empty data.

**put**
https://github.com/status-im/nim-datastore/blob/a82389743cc5e342b6ce64a9a79b025214d7572d/tests/datastore/test_sqlite_datastore.nim#L121
(need to scroll vertically in the following code-box)
https://github.com/status-im/nim-datastore/blob/a82389743cc5e342b6ce64a9a79b025214d7572d/tests/datastore/test_sqlite_datastore.nim#L157-L171
**get**
https://github.com/status-im/nim-datastore/blob/a82389743cc5e342b6ce64a9a79b025214d7572d/tests/datastore/test_sqlite_datastore.nim#L302-L313

> I think that dataCol should be NOT NULL too.
>
> Accordingly, bindParam procedure should store empty blobs (val.len == 0) as empty strings (using sqlite3_bind_blob with ptr!=NULL and len=0).

I can make this change, but I'd like to understand the motivation better. What advantages are there over `NULL` representing empty data?

> Note that code still should handle NULL values stored in any column since we may deal with DB file created by other CREATE statement.

This worries me a bit, seems like it may overcomplicate things. We already have a strict requirement that a database schema created by a different `CREATE` statement must be equivalent to or a superset of our schema (leaving aside NULL consideration), else creation of the prepared statements in `SQLiteDatastore.new` will fail and therefore object construction will fail.

Could we simply document that a user should migrate data from their own database into one created by nim-datastore if they want to work with pre-existing data?